### PR TITLE
cti-stix-ui

### PR DESCRIPTION
**Pull request by infosec-alchemist**
Initial check in of CTI-STIX-UI.

### CODE REVIEW

File: cti-stix-ui/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+COPY package.json $WORKING_DIRECTORY
+COPY bower.json $WORKING_DIRECTORY
+COPY ember-cli-build.js $WORKING_DIRECTORY
+
+RUN echo "Building cti-ui"
+
+RUN npm install
+RUN bower --allow-root install
+RUN npm rebuild node-sass
+#RUN ember build
```
Comment (exceptionfactory) [inline cti-stix-ui/Dockerfile]: Recommend removing unnecessary comments

File: cti-stix-ui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+COPY . $WORKING_DIRECTORY
+
+# ports variables
```
Comment (exceptionfactory) [inline cti-stix-ui/Dockerfile]: Recommend removing unnecessary comments

File: cti-stix-ui/app/components/phase-group-summary-card.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import Ember from "ember";
```
Comment (exceptionfactory) [inline cti-stix-ui/app/components/phase-group-summary-card.js]: This component is not used.

File: cti-stix-ui/app/services/course-of-action.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
+import Ember from 'ember';
```
Comment (exceptionfactory) [inline cti-stix-ui/app/services/course-of-action.js]: This service class is specific to reporting and should be removed.

File: cti-stix-ui/app/services/granular-marking.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import Ember from 'ember';
```
Comment (exceptionfactory) [inline cti-stix-ui/app/services/granular-marking.js]: This service class is also specific to reporting and should be removed.

File: cti-stix-ui/app/templates/components/action-switches-switch.hbs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div class="row">
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/components/action-switches-switch.hbs]: This component is unused and should be removed.

File: cti-stix-ui/app/templates/components/attack-pattern-description.hbs [hunk 7]
```diff
@@ -0,0 +1,52 @@
+<h5 class="attack-pattern-header">
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/components/attack-pattern-description.hbs]: This component is unused and should be removed.

File: cti-stix-ui/app/templates/components/header-navigation-bar.hbs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    {{#link-to "relationships" tagName="li"}}
+      {{link-to "Relationships" "relationships"}}
+    {{/link-to}}
+    {{#link-to "threat-actors" tagName="li"}}
+      {{link-to "Threat Actors" "threat-actors"}}
+    {{/link-to}}
+  </ul>
+
+  <ul class="right hide-on-med-and-down">
+    <li><a target="_blank" href="https://app.swaggerhub.com/api/unfetter/cti-stix-store">View APIs</a></li>
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/components/header-navigation-bar.hbs]: Recommend changing the label to just API.  The link could be changed to a relative path of /cti-stix-store-api/specifications/ in order to reflect what the UI is actually calling.

File: cti-stix-ui/app/controllers/course-of-action-new.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+            let labels = Ember.get(item, "labels");
+            labels.forEach(function(label) {
+                if (Ember.isPresent(label.label)) {
+                    labelsArray.push(label.label);
+                }
+            });
+            Ember.set(item, "labels", labelsArray);
+
+            let ratingLabelArray = [];
```
Comment (exceptionfactory) [inline cti-stix-ui/app/controllers/course-of-action-new.js]: Rating Labels should be removed as they are specific to unfetter-discover.

File: cti-stix-ui/app/models/course-of-action.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+ * 
+ * @module
+ * @extends ember-data/Model
+ */
+export default DS.Model.extend({
+    name: DS.attr("string"),
+    description: DS.attr(),
+    labels: DS.attr(),
+    external_references: DS.attr(),
+    x_unfetter_rating_labels: DS.attr(),
```
Comment (exceptionfactory) [inline cti-stix-ui/app/models/course-of-action.js]: Rating labels should be removed as they are specific to unfetter-discover.

File: cti-stix-ui/app/routes/course-of-action-new.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+            item: {
+                name: undefined,
+                description: undefined,
+                labels: [],
+                external_references: [],
+                created: new Date(),
+                modified: new Date(),
+                version: "1",
+                /**      Remove these later when we are sure how to handle them */          
+                x_unfetter_rating_labels: [
```
Comment (exceptionfactory) [inline cti-stix-ui/app/routes/course-of-action-new.js]: Rating labels should be removed as they are specific to unfetter-discover.

File: cti-stix-ui/app/templates/course-of-action-new.hbs [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+      </div>
+    {{/md-card-content}}
+    {{#md-card-action}}
+      {{md-btn text="Remove" action="removeLabel" actionArg=label}}
+    {{/md-card-action}}
+  {{/md-card}}
+
+{{/each}}
+
+<h5>Rating Lables</h5>
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/course-of-action-new.hbs]: Rating labels should be removed as they are specific to unfetter-discover.

File: cti-stix-ui/app/templates/course-of-action.hbs [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+        <p>
+        {{#if model.item.labels}}
+            {{#each model.item.labels as |label|}}
+              <div class="chip">{{label}}</div>
+            {{/each}}
+        {{/if}}
+        </p>
+      </div>
+    </div>
+  <div class="row">
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/course-of-action.hbs]: Rating labels should be removed as they are specific to unfetter-discover.

File: cti-stix-ui/app/templates/index.hbs [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    </div>
+  </div>
+ 
+
+
+  <footer class="application-header">
+    
+    <div class="footer-copyright">
+      <div class="container">
+      Made by <a class="white text-lighten-3" target="_blank" href="http://www.iad.gov">NSA Information Assurance</a>
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/index.hbs]: Should this be present in the footer? If so, the white background should be removed.

File: cti-stix-ui/app/templates/partners.hbs [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+        </div>
+        <div class="card-action">
+           <a href="http://www.oasis-open.org" target="_blank">Visit OASIS&reg;</a>
+        </div>
+        </div>
+    </div>
+    <div class="col s12 m6">
+        <div class="card large">
+        <div class="card-image" >
+            <img src="/cti-stix-ui/iad.png">
```
Comment (exceptionfactory) [inline cti-stix-ui/app/templates/partners.hbs]: This file does not exist and results in a broken image.

File: cti-stix-ui/app/transitions.js [hunk 16]
```diff
@@ -0,0 +1,12 @@
+export default function() {
```
Comment (exceptionfactory) [inline cti-stix-ui/app/transitions.js]: Transitions are not used outside of reports and this file can be removed.

File: cti-stix-ui/bower.json [hunk 17]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "cti-sitx-ui",
```
Comment (exceptionfactory) [inline cti-stix-ui/bower.json]: The name property should be cti-stix-ui.

File: cti-stix-ui/package.json [hunk 18]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "cti-stix-ui",
+  "version": "0.1.1",
+  "description": "Unfetter Discover UI",
```
Comment (exceptionfactory) [inline cti-stix-ui/package.json]: The description should be updated.

File: cti-stix-ui/tests/index.html [hunk 19]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <title>UnfetterDiscoverUi Tests</title>
```
Comment (exceptionfactory) [inline cti-stix-ui/tests/index.html]: The title tag should be changed.

File: cti-stix-ui/tests/integration/components/relationships-graph-test.js [hunk 20]
```diff
@@ -0,0 +1,24 @@
+import { moduleForComponent, test } from 'ember-qunit';
```
Comment (exceptionfactory) [inline cti-stix-ui/tests/integration/components/relationships-graph-test.js]: This file can be removed.

File: cti-stix-ui/Dockerfile (hunk 2 above)
Comment (infosec-alchemist) [inline cti-stix-ui/Dockerfile]: why would this be considered unnecessary?

File: cti-stix-ui/app/templates/components/header-navigation-bar.hbs (hunk 8 above)
Comment (infosec-alchemist) [inline cti-stix-ui/app/templates/components/header-navigation-bar.hbs]: Okay, I've changed it.

**Pull request closed without merging**